File: app/lib/supabase_test.dart
import 'supabase_config.dart';

class SupabaseTest {
  static Future<Map<String, dynamic>> testConnection() async {
    try {
      final response = await SupabaseConfig.client
          .from('test_table')
          .select('*')
          .limit(1);

      return {
        'success': true,
        'message':
            'Connected to Supabase (table doesn\'t exist yet - this is expected)',
        'data': response,
      };
    } catch (err) {
      return {'success': false, 'error': err.toString()};
    }
  }

  static Future<Map<String, dynamic>> testAuth() async {
    try {
      final session = SupabaseConfig.client.auth.currentSession;
      return {'success': true, 'authenticated': session != null};
    } catch (err) {
      return {'success': false, 'error': err.toString()};
    }
  }
}

File: web/src/supabase-test.js
import { supabase } from './supabase.js'

// Test Supabase connection
export async function testSupabaseConnection() {
  try {
    const { data, error } = await supabase
      .from('test_table')
      .select('*')
      .limit(1)
    
    if (error) {
      console.log('Supabase connected successfully (table doesn\'t exist yet - this is expected)')
      return { success: true, message: 'Connected to Supabase' }
    }
    
    return { success: true, data }
  } catch (err) {
    console.error('Supabase connection error:', err)
    return { success: false, error: err.message }
  }
}

// Test authentication
export async function testAuth() {
  try {
    const { data: { session }, error } = await supabase.auth.getSession()
    return { success: true, session: !!session }
  } catch (err) {
    return { success: false, error: err.message }
  }
}
